search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+ * Search API and passes them to be saved in countryData object.
+ * @param {string} countryCode 2 letter country code for search results to be
+ *     written in.
+ * @param {string} query Search query.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function getSearchResultsForCountryFromAPI(countryCode, query, countryData) {
+  let response =
+    await fetch('https://www.googleapis.com/customsearch/v1?key=AIzaSyDszWv1aGP7Q1uOt74CqBpx87KpkhDR6Io&cx=017187910465527070415:o5pur9drtw0&q=' + query + '&cr=' + countryCode + '&num=10&safe=active&dateRestrict=d1&fields=items(title,snippet,htmlTitle,link)');
+  let searchResults = await response.json();
+  return await saveResultsAndDeletePrevious(searchResults, countryData);
+}
+
+/**
+ * Deletes previous search results and saves the current results in the
+ * Datastore, given the results JSON obtained from the API.
+ * @param {Object} searchResultsJson Object with information for top 10 search
+ *     results.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function saveResultsAndDeletePrevious(searchResultsJson, countryData) {
+  await deleteAncientResults();
+
+  // Parse the JSON string and pass each search result to add to the
+  // countryData object.
+  var currentSearchResults = searchResultsJson.items;
+  try {
+    let avg = 0;
+    if (currentSearchResults == undefined) {
+      return 0;
+    } else {
+      for (var i = 0; i < currentSearchResults.length; i++) {
+        avg = avg + await addSearchResultToCountryData(currentSearchResults[i], countryData);
+        console.log('avg in for loop of saveResults' + avg);
+      }
+      return avg / currentSearchResults.length;
+    }
+  } catch (err) { // Occurs when no search results for that country and topic.
+    console.error('ERROR:', err);
+    countryData = null;
+  }
+}
+
+/** Deletes search results from 7 days ago. */
+async function deleteAncientResults() {
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp');
+  const [searchResults] = await datastore.runQuery(query);
+
+  // Note: Can't use forEach with await.
+  // Loop through sorted data beginnning with oldest results, delete if older
+  // than a week. Stop when reach results from within a week.
+  for (let i = 0; i < searchResults.length; i++) {
+    if (Date.now() - searchResults[i].timestamp > 7 * 24 * 60 * 60000) {
+      const searchResultKey = searchResults[i][datastore.KEY];
+      await datastore.delete(searchResultKey);
+      console.log(`Custom Search Result ${searchResultKey.id} deleted.`)
+    } else {
+      break;
+    }
+  }
+}
+
+/** 
+ * Creates Datastore item for given topic and country search results. 
+ * @param {string} topic The seach topic the search results are for.
+ * @param {Object} countriesData Object holding all searchResults for all
+ *      countries.
+ */
+async function addTopicToDatastore(topic, countriesData) {
```

Comment:
`addTopicDataToDataStore`

Reply:
Personally, I think this method name is sufficient, because it is implied with Datastore. @carmenBen You guys can discuss further, but I would keep this method name.

Reply:
I agree, I think it is fine to be honest.

Reply:
But you are not adding a topic. You are adding data associated with the topic

Reply:
I agree with Chenyu. You're adding more than just the topic. But TopicData woudn't be too accurate either, since you have more than just that as well.

Reply:
How about addCustomSearchTopicToDatastore? The name of the entity type is Custom Search Topic.

Or addCustomSearchTopicEntityToDatastore?

Reply:
addCustomSearchTopicEntityToDatastore sounds good to me